### PR TITLE
Fixed performance by avoiding WC_Product::get_available_variations().

### PR DESCRIPTION
**Pull request by bogdanarizancu**
### Ticket
- [Performance: Optimize trusted shops plugin](https://app.asana.com/0/1162879836389371/1116431680117185)

### CODE REVIEW

File: src/TrustedShops.php [hunk 1]
```diff
@@ -225,10 +225,18 @@ public static function wp_footer() {
...
+      $variation_ids = $product->get_visible_children();
+      $placeholders = implode(',', array_fill(0, count($variation_ids), '%d'));
+      $products_sku = $wpdb->get_col($wpdb->prepare(
+        "SELECT pm.meta_value AS attachment_id
+        FROM wp_posts p
+        INNER JOIN wp_postmeta pm ON pm.post_id = p.ID AND pm.meta_key = '_sku'
+        WHERE p.ID IN ($placeholders)",
+        $variation_ids
+      ));
+      error_log(print_r(count($variation_ids), TRUE));
```
Comment (sun) [inline src/TrustedShops.php:239]: ?

Comment (bogdanarizancu) [reply src/TrustedShops.php:239]: Dumb mistake, thanks.

File: src/TrustedShops.php [hunk 2]
```diff
@@ -225,10 +225,19 @@ public static function wp_footer() {
 
     $products_sku = [];
     if ($product->get_type() === 'variable') {
-      $variations = $product->get_available_variations();
-      foreach ($variations as $variation) {
-        $products_sku[] = $variation['sku'] ?? '';
-      }
+      // Gets variations skus, avoiding "get_available_variations()".
```
Comment (sun) [inline src/TrustedShops.php:228]:
```suggestion
      // Avoid WC_Product_Variable::get_available_variations() as it additionally
      // produces a lot of output for templates, which is not necessary here.
```

File: src/TrustedShops.php [hunk 3]
```diff
@@ -225,10 +225,19 @@ public static function wp_footer() {
...
+      global $wpdb;
```
Comment (sun) [inline src/TrustedShops.php:229]: Globals should be imported only once at the beginning of the function - please move this accordingly.

File: src/TrustedShops.php [hunk 4]
```diff
@@ -225,10 +225,19 @@ public static function wp_footer() {
...
+      $placeholders = implode(',', array_fill(0, count($variation_ids), '%d'));
+      $products_sku = $wpdb->get_col(
+        $wpdb->prepare(
+          "SELECT pm.meta_value AS attachment_id
+          FROM wp_posts p
+          INNER JOIN wp_postmeta pm ON pm.post_id = p.ID AND pm.meta_key = '_sku'
+          WHERE p.ID IN ($placeholders)",
+          $variation_ids
+        )
+      );
```
Comment (sun) [inline src/TrustedShops.php:240]: Not sure why you are joining on posts here… - am I missing something?

And what's `attachment_id` doing here? ;-)

Please also follow the following code style:
```suggestion
      $products_sku = $wpdb->get_col($wpdb->prepare("
        SELECT pm.meta_value AS sku
        FROM wp_postmeta pm
        WHERE pm.post_id IN ($placeholders) AND pm.meta_key = '_sku'
      ", $variation_ids));
```

Comment (bogdanarizancu) [reply src/TrustedShops.php:240]: I must have been really tired :), thanks.

**Comment by sun** [comment]
Please use the corrected PR title as commit message.

**Pull request merged**